redfish-core/lib/assembly.hpp

```diff
@@ -0,0 +1,494 @@
+        "xyz.openbmc_project.Inventory.Item.Battery",
+        "xyz.openbmc_project.Inventory.Item.DiskBackplane"};
+
+    std::size_t assemblyIndex = 0;
+
+    for (const auto& assembly : assemblies)
+    {
+        nlohmann::json& tempyArray = asyncResp->res.jsonValue["Assemblies"];
+
+        boost::urls::url dataID = boost::urls::format(
+            "/redfish/v1/Chassis/{}/Assembly#/Assemblies/{}", chassis,
+            std::to_string(assemblyIndex));
+
+        nlohmann::json::object_t item;
+        item["@odata.type"] = "#Assembly.v1_3_0.AssemblyData";
+        item["@odata.id"] = dataID;
+        item["MemberId"] = std::to_string(assemblyIndex);
+
+        tempyArray.emplace_back(item);
+
+        tempyArray.at(assemblyIndex)["Name"] =
+            sdbusplus::message::object_path(assembly).filename();
+
+        dbus::utility::getDbusObject(
+            assembly, interfaces,
+            [asyncResp, assemblyIndex,
+             assembly](const boost::system::error_code ec,
+                       const dbus::utility::MapperGetObject& object) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("DBUS response error : {}", ec.value());
+                messages::internalError(asyncResp->res);
+                return;
+            }
+
+            for (const auto& [serviceName, interfaceList] : object)
+            {
+                for (const auto& interface : interfaceList)
+                {
+                    if (interface ==
+                        "xyz.openbmc_project.Inventory.Decorator.Asset")
+                    {
+                        getAssemblyAsset(asyncResp, serviceName, assembly,
+                                         assemblyIndex);
+                    }
+                    else if (interface == "xyz.openbmc_project.Inventory."
+                                          "Decorator.LocationCode")
+                    {
+                        getAssemblyLocationCode(asyncResp, serviceName,
+                                                assembly, assemblyIndex);
+                    }
+                }
+            }
+        });
+
+        nlohmann::json::array_t assemblyArray =
+            asyncResp->res.jsonValue["Assemblies"];
+        asyncResp->res.jsonValue["Assemblies@odata.count"] =
+            assemblyArray.size();
+
+        assemblyIndex++;
+    }
+}
+
+/**
+ * @brief Api to check if the assemblies fetched from association Json is
+ * also implemented in the system. In case the interface for that assembly
+ * is not found update the list and fetch properties for only implemented
+ * assemblies.
+ * @param[in] asyncResp - Shared pointer for asynchronous calls.
+ * @param[in] chassisPath - Chassis the assemblies are associated with.
+ * @param[in] assemblies - list of all the assemblies associated with the
+ * chassis.
+ * @return None.
+ */
+inline void
+    checkAssemblyInterface(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
```

Comment:
I think some of  functions have been moved to `redfish-core/include/utils/chassis_utils.hpp` (in 1060).
- checkAssemblyInterface
- checkAssociation
- getChassisAssembly

Should we put them into `redfish-core/include/utils/chassis_utils.hpp` ?

Reply:
Understood. Thanks